postgis/lwgeom_functions_basic.c

```diff
@@ -2399,6 +2400,88 @@ Datum LWGEOM_azimuth(PG_FUNCTION_ARGS)
 	PG_RETURN_FLOAT8(result);
 }
 
+/**
+ * Compute the angle defined by 3 points or the angle between 2 vectors
+ * defined by 4 points
+ * given Point geometries.
+ * @return NULL on exception (same point).
+ * 		Return radians otherwise (always positive).
+ */
+PG_FUNCTION_INFO_V1(LWGEOM_angle);
+Datum LWGEOM_angle(PG_FUNCTION_ARGS)
+{
+	GSERIALIZED *geom;
+	LWGEOM *geom_unser;
+	LWPOINT *lwpoint;
+	POINT2D points[4]; 
+	double az1,az2 ;
+	double result;
+	int srids[4];
+	int i = 0 ;  
+	int n_args = 0; 
+
+	/* extract points while it is possible*/
+	n_args = PG_NARGS() ;
+	for(i=0; i<n_args; i++)
+	{
+		geom = PG_GETARG_GSERIALIZED_P(i);
+		geom_unser = lwgeom_from_gserialized(geom) ; 
+		if (lwgeom_is_empty(geom_unser) == true)
+		{/*4th arg is not set, don't use it*/
+			n_args = i ; 
+			if (n_args < 3)
+				lwpgerror("Empty geometry"); 
+			break; 
+		}
+		
+		lwpoint = lwgeom_as_lwpoint(geom_unser);
+		if ( ! lwpoint )
+		{
+			PG_FREE_IF_COPY(geom, i);
```

Comment:
These memory releases should really affect all the previous geometry inputs, or only the last one would be released.